packages/ifpack2/test/unit_tests/Ifpack2_UnitTestOverlappingRowMatrix.cpp

```diff
@@ -184,14 +184,14 @@ void localReducedMatvec(const MatrixClass & A_lcl,
         // NOTE: It looks like I should be able to get this data up above, but if I try to
         // we get internal compiler errors.  Who knew that gcc tried to "gimplify"?
         const LO numVectors = static_cast<LO>(X_lcl.extent(1));
-        Kokkos::parallel_for(Kokkos::TeamThreadRange (dev, 0, rows_per_team),[&] (const LO& loop) {
+        Kokkos::parallel_for(Kokkos::TeamThreadRange (dev, 0, rows_per_team),[&] (const LO loop) {
             const LO lclRow = static_cast<LO> (dev.league_rank ()) * rows_per_team + loop;
             
             if (lclRow >= numLocalRows) {
               return;
             }
             const auto A_row = A_lcl.rowConst(lclRow);
-            const LO row_length = static_cast<LO> (A_row.length);
+            const LO row_length = A_row.length;
             for(LO v=0; v<numVectors; v++) {
               residual_value_type A_x = KAT::zero ();          
```

Comment:
I'm realizing now that we really don't need `ArithTraits` for zero -- `{}` initialization works just fine for any reasonable arithmetic type:
```c++
residual_value_type A_x {};
```

Reply:
This is a strong argument for consolidating the "almost matvec" kernels in one place... KAT is used in the other two or three places these guys exist...